AdyenDropIn/Components/PaymentMethodListComponent.swift

```diff
@@ -38,14 +45,34 @@ internal final class PaymentMethodListComponent: ComponentLoader, PresentableCom
         self.style = style
     }
     
+    internal func reload(with components: [ComponentsSection]) {
+        componentSections = components
+        listViewController.reload(newSections: createListSections())
+    }
+    
+    internal func deleteComponent(at indexPath: IndexPath) {
+        componentSections.deleteItem(at: indexPath)
+        listViewController.deleteItem(at: indexPath)
+    }
+    
     // MARK: - View Controller
     
     /// :nodoc:
     internal var viewController: UIViewController { listViewController }
 
     private let brandProtectedComponents: Set = ["applepay"]
     
-    internal lazy var listViewController: ListViewController = {
+    internal lazy var listViewController: ListViewController = createListViewController()
+    
+    private func createListViewController() -> ListViewController {
+        let listViewController = ListViewController(style: style)
+        listViewController.title = localizedString(.paymentMethodsTitle, localizationParameters)
+        listViewController.reload(newSections: createListSections())
+        
+        return listViewController
+    }
+    
+    private func createListSections() -> [ListSection] {
         func item(for component: PaymentComponent) -> ListItem {
```

Comment:
can this function be moved outside? a bit unusual to see in this project. then we can also drop `return` at line 99